adr/2025-05028-authorisation-model.md

```diff
@@ -0,0 +1,150 @@
+
+![Access Control Model](/images/userrole-model.png)
+
+In the example model above, Roles are a set list controlled by an enum which can easily be extended but keeps roles to a known quantity. Having a separate table for Roles was also considered, but initially while our set of Roles are relatively few, a separate table felt slightly over-engineered.  
+
+You then have the ability to define roles and permissions by using the nullable columns:
+
+| Role type | Description | Database row |
+| --- | --- | --- |
+| Funding Service administrator | Full platform admin priveleges | ADMIN role with no `organisation_id` or `grant_id` |
+| MHCLG Organisation admin | An internal user who needs full overview of all MHCLG grants | ADMIN role with no `grant_id` |
+| MHCLG Grant team member | An internal Grant Team user who read-only access to a specific grant | MEMBER role with an `organisation_id` and `grant_id` (`organisation_id` potentially optional) |
+| External Organisation admin | An external user who needs full overview of that Organisation's applications/monitoring returns | ADMIN role with no `grant_id` |
+| Organisation member | A user who needs read-only access to everything attached to an organisation | MEMBER role with no `grant_id` |
+| Grant admin | A user who needs top-level priveleges for a specific grant, either internal eg. grant team admin or external eg. someone with oversight of an entire application or monitoring return | ADMIN role with both an `organisation_id` and `grant_id` |
+| Organisation S151 Officer | An Organisation's S151 Officer | S151_OFFICER role with an `organisation_id` |
+| Grant editor | An internal user who will need to edit Grant details or Grant collection information, or an external user who can edit parts of a grant application or monitoring return | EDITOR role with both an `organisation_id` and `grant_id` |
+| Grant Assessor | An internal user assigned to assess individual grant applications or review other submitted data | ASSESSOR role with an `organisation_id` and `grant_id`|
+
+This also allows developers to produce easy, reusable checks for permissions levels and only need to check one table based on the active user's information, and also implement database-level constraints on certain combinations which should not be allowd, eg. a MEMBER should not be possible if `organisation_id` and `grant_id` are `null` or `S151_OFFICER` should only be possible with an `organisation_id` and no `grant_id` ie. they are a S151_OFFICER at an Organisation level.
+
+It's worth noting why MHCLG is defined as a distinct Organisation in the example table above. This is predominantly in the anticipation that OGDs will be using the service at some point, and there will be a need to define permissions at Organisation-level. If the service was remaining MHCLG-only then this wouldn't be necessary. 
```

Comment:
I am currently thinking that MHCLG shouldn't be treated as some kind of special organisation, but rather than arbitrary organisations can have the ability to create/own grants. MHCLG will then just be the first instance of that, but eventually OGDs will have that state/role as well.

This way we can remove the "uniqueness" of MHCLG and turn it into a platform feature instead of something that gets hardcoded/given special snowflake status 👀  bit of WIP thinking though.